controllers/openshift_controller.go

```diff
@@ -1016,7 +1018,7 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigInstallRequest() (ctrl.
 	if mcfgv1.IsMachineConfigPoolConditionTrue(foundMcp.Status.Conditions, mcfgv1.MachineConfigPoolUpdated) &&
 		foundMcp.Status.UpdatedMachineCount == foundMcp.Status.MachineCount {
 		r.Log.Info("create runtime class")
-		r.kataConfig.Status.InstallationStatus.IsInProgress = "false"
+		r.kataConfig.Status.InstallationStatus.IsInProgress = corev1.ConditionFalse
```

Comment:
True... I didn't do that though as I knew I was going to remove the line altogether (removing conditionals like the one you point to is one of the main goals of this PR :-)).

Anyway, I suspect the real fix would be to change the type of `r.kataConfig.Status.InstallationStatus.IsInProgress` (and likely others in `KataConfig.status`) to boolean.  `corev1.ConditionStatus` is really a tri-state (True, False and Unknown IIRC), probably primarily meant for k8s Conditions where k8s might legimately not know if a condition is true or false.

Our fields like `IsInProgress` cannot really benefit from this and a simple boolean feels therefore more appropriate.  Moreover, `corev1.ConditionStatus` is really just another name for `string` so any mistyped string can be compared or assigned to it without the compiler noticing so that would be in my view another reason to use it only when necessary.

Reply:
> True... I didn't do that though as I knew I was going to remove the line altogether (removing conditionals like the one you point to is one of the main goals of this PR :-)).
> 

Well... this obviously breaks bisect, which has always been considered a capital sin in my previous assignment :wink: . This being said, since I haven't had the opportunity to bisect on the operator so far, maybe it isn't a big deal.

> Anyway, I suspect the real fix would be to change the type of `r.kataConfig.Status.InstallationStatus.IsInProgress` (and likely others in `KataConfig.status`) to boolean. `corev1.ConditionStatus` is really a tri-state (True, False and Unknown IIRC), probably primarily meant for k8s Conditions where k8s might legimately not know if a condition is true or false.
> 
> Our fields like `IsInProgress` cannot really benefit from this and a simple boolean feels therefore more appropriate. Moreover, `corev1.ConditionStatus` is really just another name for `string` so any mistyped string can be compared or assigned to it without the compiler noticing so that would be in my view another reason to use it only when necessary.

I tend to agree that it doesn't make sense to have an unknown state for something that is controlled by the operator and thus always known.

Reply:
> Well... this obviously breaks bisect, which has always been considered a capital sin in my previous assignment wink . This being said, since I haven't had the opportunity to bisect on the operator so far, maybe it isn't a big deal.

I'm not sure if I follow correctly but it would seem to me that it doesn't really _break_ it at least in the sense that it doesn't break it more than it already was.

Reply:
You're right in general, I agree.  In this specific case though, I think that there still would be no breakage since if you take a look at the block of code guarded by that conditional, it only
- sets `IsInProgress` to `true` which is superfluous at that point (it was done in `createExtensionMc()` already)
- requeues for 15 seconds which is superflous too. Let me explain: if the `IsInProgress == "false"` test was the _only_ reason the conditional we're discussing wouldn't trigger (which appears implicitly assumed - otherwise there would be nothing to discuss ;-)) then the MCP in question is `Updating`.  In that case however, the conditional right after this one will _not_ trigger as it requires the MCP in `Updated` so its `else` branch would be executed which... requeues for 15 seconds. ;-)

So I think that while I cannot prove that there's no effect on the check, assuming the above is correct I might be able to prove that there's no effect on code execution anyway.

I do agree though that it was kind of reckless of me.  BTW I think we should discuss the standards required of PRs explicitly at some point.  I've been trying to stick to the standard you use (functionality shouldn't be harmed by any commit) but I'm noticing that if we relaxed the requirement to e.g. every commit has just to be compilable it would make it possible to split some of the bigger commits (like the final one in this series) into a bunch of much more understandable & reviewable chunks.

Reply:
`IsInProgress` is in `KataConfig.status`, not in an MC, but I get your point.

Well the original controller, before I set out to change it, only handled initial installation and subsequent uninstallation, pretty much no effort was made to handle updating an existing installation (by adding or removing kata-enabled nodes).  This is kind of a breakage (by omission) by itself but if we accept it it makes sense only to set `InstalllationStatus.IsInProgress` when the extension MC is created.  So yes, if the MC already existed `IsInProgress` wouldn't be set - but if the MC already existed (at the beginning of an installation reconciliation, as opposed to in the middle of the initial installation when `IsInProgress` is already set from the time of the MC creation) that means that we're performing an update.  But update was already broken in the context of the original controller so no additional breakage here.

Now, since my recent changes which fixed and enabled installation updating, this could have mattered since updating is not broken anymore, except that `IsInProgress` is actually only used by status reporting (`updateStatus()` and its callees). But while the original support for updating was rudimentary, the corresponding support for update progress reporting was outright missing, non-existent, stubbed-out, and unlike the updating proper which has been fixed recently, status reporting hasn't changed yet.

So all in all, wrong handling of `IsInProgress` in the context of installation update basically breaks code that doesn't even exist yet.

Reply:
Oops sorry about the confusion with MC but fortunately you understood :wink:

Thanks for the detailed explanation, it helps.